src/portfft/descriptor.hpp

```diff
@@ -231,16 +231,23 @@ class committed_descriptor {
   };
 
   struct dimension_struct {
-    std::vector<kernel_data_struct> kernels;
+    std::vector<kernel_data_struct> forward_kernels;
+    std::vector<kernel_data_struct> backward_kernels;
+    std::vector<kernel_data_struct> transpose_kernels;
     std::shared_ptr<IdxGlobal> factors_and_scan;
     detail::level level;
     std::size_t length;
     Idx used_sg_size;
     Idx num_batches_in_l2;
     Idx num_factors;
 
-    dimension_struct(std::vector<kernel_data_struct> kernels, detail::level level, std::size_t length, Idx used_sg_size)
-        : kernels(kernels), level(level), length(length), used_sg_size(used_sg_size) {}
+    dimension_struct(std::vector<kernel_data_struct> forward_kernels, std::vector<kernel_data_struct> backward_kernels,
+                     detail::level level, std::size_t length, Idx used_sg_size)
```

Comment:
`transpose_kernels`?

Reply:
Constructor of `dimension_struct` will only take parameters which will be required for all implementations.
Since the `transpose_kernels` are only required for global, that vector is populated separately